evaluate_adapter.py

```diff
@@ -0,0 +1,164 @@
+    return testdata
+
+
+def main(
+    datasets: str = "wikitext,ptb,c4",
+    *,
+    # compilation fails as it does not support torch.complex64 for RoPE
+    # compile: bool = False,
+    accelerator: str = "auto",
+    adapter_path: Optional[Path] = None,
+    checkpoint_path: Optional[Path] = None,
+    tokenizer_path: Optional[Path] = None,
+    dtype: str = "float32",
+    quantize: Optional[str] = None,
+) -> None:
+    """Generates text samples based on a pre-trained LLaMA model and tokenizer.
+
+    Args:
+        datasets: The datasets to use as a comma separated string
+        # compile: Whether to compile the model.
+        accelerator: The hardware to run on. Possible choices are:
+            ``"cpu"``, ``"cuda"``, ``"mps"``, ``"gpu"``, ``"tpu"``, ``"auto"``.
+        adapter_path: Path to the checkpoint with trained adapter weights, which are the output of
+            `finetune_adapter.py`.
+        checkpoint_path: The checkpoint path to load.
+        tokenizer_path: The tokenizer path to load.
+        quantize: Whether to quantize the model and using which method:
+            ``"llm.int8"``: LLM.int8() mode,
+            ``"gptq.int4"``: GPTQ 4-bit mode.
+    """
+    if not adapter_path:
+        adapter_path = Path("out/adapter/alpaca/lit-llama-adapter-finetuned.pth")
+    if not checkpoint_path:
+        checkpoint_path = Path(f"./checkpoints/lit-llama/7B/lit-llama.pth")
+    if not tokenizer_path:
+        tokenizer_path = Path("./checkpoints/lit-llama/tokenizer.model")
+    
+    assert adapter_path.is_file()
+    assert checkpoint_path.is_file()
+    assert tokenizer_path.is_file()
+
+    fabric = L.Fabric(accelerator=accelerator, devices=1)
+
+    dt = getattr(torch, dtype, None)
+    if not isinstance(dt, torch.dtype):
+        raise ValueError(f"{dtype} is not a valid dtype.")
+    dtype = dt
+
+    with EmptyInitOnDevice(
+        device=fabric.device, dtype=dtype, quantization_mode=quantize
+    ):
+        print("Loading model ...", file=sys.stderr)
+        t0 = time.time()
+        pretrained_checkpoint = lazy_load(checkpoint_path)
+        adapter_checkpoint = lazy_load(adapter_path)
+        name = llama_model_lookup(pretrained_checkpoint)
+        model = LLaMA.from_name(name)
+
+        # 1. Load the pretrained weights
+        model.load_state_dict(pretrained_checkpoint, strict=False)
+        # 2. Load the fine-tuned adapter weights
+        model.load_state_dict(adapter_checkpoint, strict=False)
+
+        print(f"Time to load model: {time.time() - t0:.02f} seconds.", file=sys.stderr)
+
+    model.eval()
+
+    # if compile:
+    #     model = torch.compile(model)
+
+    total_toks = 0
+    model = fabric.setup_module(model)
+
+    tokenizer = Tokenizer(tokenizer_path)
+
+    for dsname in datasets.split(","):
+        test_string = load_eval_data(dsname)
+
+        sample = {"instruction": test_string, "input": input}
+        test_string = generate_prompt(sample)
```

Comment:
@rasbt In the PR description, you wrote that these lines could be removed. Do you mean that the encoding into the instruction template is not necessary, and the prompt can be given as is? But how would the instructiono be combined with the optional "input"?

Reply:
@awaelchli 

Yeah, so what I meant was instead of

```python
       test_string = load_eval_data(dsname)

         sample = {"instruction": test_string, "input": input}
         test_string = generate_prompt(sample)

         encoded_text = tokenizer.encode(
             test_string, bos=True, eos=False, device=fabric.device
         )
```

we can simply do

```python
       test_string = load_eval_data(dsname)
       encoded_text = tokenizer.encode(
             test_string, bos=True, eos=False, device=fabric.device
       )
```

Since we are only evaluating perplexity, I think it doesn't matter so much here. We are basically just checking if the model can generate "likely" text. So, in other words, it's not really evaluating the instruction-output pair yet. That would be some future todo item for more comprehensive Q&A evaluation.

So maybe we should leave it as

```python
       test_string = load_eval_data(dsname)

         sample = {"instruction": test_string, "input": input}
         test_string = generate_prompt(sample)

         encoded_text = tokenizer.encode(
             test_string, bos=True, eos=False, device=fabric.device
         )
```

to check whether the model can handle this format though?

Reply:
Okay, sounds good. If you remove it, could you add a small comment in that line that the conversion to the template is missing and why? Thanks!

Reply:
Oh I would maybe keep it because it's the "more complete" code. What do you think?

Reply:
Yep, good with me. Good to merge this then?

Reply:
Yes would be good to merge. (Btw I compared the two approaches mentioned above, and the performance was basically the same. In some cases there was a tiny 0.01 point difference which doesn't really matter in practice)